File: src/FestivalPOS/src/api/account.ts
import { Big } from "big.js";
import { bigConverter } from "./converters/big-converter";
import { jsonProperty } from "@utiliread/json";

export class Account {
    @jsonProperty()
    id!: number;

    @jsonProperty()
    number!: number;

    @jsonProperty()
    name!: string;

    @jsonProperty({ converter: bigConverter })
    maxCredit!: Big;

    @jsonProperty({ converter: bigConverter })
    remainingCredit!: Big;

    @jsonProperty()
    highPriorityServing!: boolean;
}

File: src/FestivalPOS.Models/Account.cs
using System.ComponentModel.DataAnnotations;

namespace FestivalPOS.Models;

public class Account
{
    public int Id { get; set; }
    public int Number { get; set; }

    [MaxLength(100)]
    public required string Name { get; set; }
    public decimal MaxCredit { get; set; }
    public decimal RemainingCredit { get; set; }
    public bool HighPriorityServing { get; set; }
    public bool IsDeleted { get; set; }
    public ICollection<Payment> Payments { get; set; } = [];
}
